service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcDocumentTaskExecutor.java

```diff
@@ -43,12 +44,20 @@ public void execute(GetGpcDocumentTaskDefinition taskDefinition) {
         String documentName = taskDefinition.getDocumentId() + JSON_EXTENSION;
         String taskId = taskDefinition.getTaskId();
 
-        var storageDataWrapper = StorageDataWrapperProvider.buildStorageDataWrapper(taskDefinition, response, taskId);
-        storageConnectorService.uploadFile(storageDataWrapper, documentName);
+        var storageDataWrapperWithDocumentRecord = StorageDataWrapperProvider.buildStorageDataWrapper(taskDefinition, response, taskId);
+        storageConnectorService.uploadFile(storageDataWrapperWithDocumentRecord, documentName);
```

Comment:
As per @msolecka 's comment below we should only be uploading the final translated version. Remove this upload. It is not needed. The second call to uploadFile will overwrite this data anyway.